pandas/_libs/lib.pyx

```diff
@@ -1500,7 +1500,7 @@ cdef class Validator:
                                   f'must define is_value_typed')
 
     cdef bint is_valid_null(self, object value) except -1:
-        return value is None or util.is_nan(value)
+        return value is None or value is C_NA or util.is_nan(value)
```

Comment:
I suppose this is to have inferring and the validation in StringArray working with NA?

One thing I have been thinking about is that it could be an option to let `pd.NA` play actually a somewhat different role than np.nan or None in construction / type inference. Eg so that if someone does `pd.Series([1, 2, pd.NA])` it automatically becomes a nullable integer dtype instead of float (or object). Since pd.NA is new, we can actually do this without breaking backwards compatibility. 
(now, not sure if that idea relates to the code here, and it can also be done later)

Reply:
> I suppose this is to have inferring and the validation in StringArray working with NA?

Correct.

> let pd.NA play actually a somewhat different role than np.nan or None in construction / type inference.

That may be in `infer_dtype`